File: RecrutementFront/RecrutementBackOffice/src/app/service/candidature.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Candidature } from '../models/candidature.model';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class CandidatureService {
  private apiUrl='http://localhost:8082/api/candidatures';

  constructor(private http: HttpClient) {}

  ajouterCandidature(candidatureRequest: Candidature): Observable<Candidature> {
    return this.http.post<Candidature>(`${this.apiUrl}`, candidatureRequest);
  }

  modifierCandidature(candidatureRequest: Candidature): Observable<Candidature> {
    return this.http.put<Candidature>(`${this.apiUrl}`, candidatureRequest);
  }

  supprimerCandidature(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }

  afficherCandidatureParId(id: number): Observable<Candidature> {
    return this.http.get<Candidature>(`${this.apiUrl}/${id}`);
  }

  afficherCandidatureParIdOffer(idOffer: number): Observable<Candidature[]> {
    return this.http.get<Candidature[]>(`${this.apiUrl}/offer/${idOffer}`);
  }

  afficherCandidature(): Observable<Candidature[]> {
    return this.http.get<Candidature[]>(`${this.apiUrl}`);
  }
}

File: RecrutementBack/src/main/java/tn/tritux/pfe/recrutement/service/CandidatureService.java
package tn.tritux.pfe.recrutement.service;

import tn.tritux.pfe.recrutement.dto.request.CandidatureRequest;
import tn.tritux.pfe.recrutement.dto.response.CandidatureResponse;

import java.util.List;

public interface CandidatureService {

    CandidatureResponse ajouterCandidature(CandidatureRequest candidatureRequest);

    CandidatureResponse modifierCandidature(CandidatureRequest candidatureRequest);
    void supprimerCandidature(Long id);
    CandidatureResponse afficherCandidatureParId(Long id);
    List<CandidatureResponse> afficherCandidature();
    List<CandidatureResponse> afficherCandidatureParIdOffer(Long idOffer);


}
